syntax = "proto3";

service AccountServiceGrpc {
    rpc getAccountStatement(AccountRequest) returns (AccountResponse);
}

message AccountRequest {
    string customer_id = 1;
}

message AccountResponse {
    string customer_name = 1;
    repeated Transaction transactions = 2;
}

message Transaction {
    string account_id = 1;
    string date = 2;
    float amount = 3;
    string transaction_type = 4;
}
